lib/elixir/lib/exception.ex

```diff
@@ -637,8 +637,16 @@ defmodule UndefinedFunctionError do
   end
 
   def message(%{reason: :"function not exported",  module: module, function: function, arity: arity, exports: exports}) do
-    "function " <> Exception.format_mfa(module, function, arity) <>
-    " is undefined or private" <> did_you_mean(module, function, arity, exports)
+    prefix = "function " <> Exception.format_mfa(module, function, arity) <> " is undefined or private"
+    suffix =
+      if macro_with_same_mfa?(module, function, arity) do
```

Comment:
Why would it load the module? Neither `function_exported?/3` nor `macro_exported?/3` load the argument module (and return `false` if not loaded) so we shouldn't end up loading anything. Does that sound right?